src/tests/ftest/io/llnl.py

```diff
@@ -0,0 +1,140 @@
+
+        # setup the DAOS python API
+        self.context = DaosContext(build_paths['PREFIX'] + '/lib/')
+
+        self.hostlist_servers = self.params.get("test_servers", '/run/hosts/')
+        self.hostfile_servers = WriteHostFile.WriteHostFile(self.hostlist_servers,
+                                                            self.workdir)
+        print("Host file servers is: {}".format(self.hostfile_servers))
+
+        self.hostlist_clients = self.params.get("test_clients", '/run/hosts/')
+        self.hostfile_clients = WriteHostFile.WriteHostFile(self.hostlist_clients,
+                                                            self.workdir, None)
+        print("Host file clients is: {}".format(self.hostfile_clients))
+
+        # start servers
+        ServerUtils.runServer(self.hostfile_servers, self.server_group,
+                              self.basepath)
+
+    def tearDown(self):
+        ServerUtils.stopServer(hosts=self.hostlist_servers)
+
+    def test_llnl(self):
+        """
+        Test ID: DAOS-2231
+        Run LLNL test provided in mpich package
+        Testing various I/O functions provided in llnl test suite
+        :avocado: tags=mpio,llnl
+        """
+        # setting llnl parameters
+        llnl_test_repo = self.params.get("llnl_repo", '/run/llnl/')
+
+        # initialize MpioUtils
+        self.mpio = MpioUtils()
+        if self.mpio.mpich_installed(self.hostlist_clients) is False:
+            self.fail("Exiting Test: Mpich not installed")
+
+        # parameters used in pool create
+        createmode = self.params.get("mode", '/run/pool/createmode/*/')
+        createuid = os.geteuid()
+        creategid = os.getegid()
+        createsetid = self.params.get("setname", '/run/pool/createset/')
+        createsize = self.params.get("size", '/run/pool/createsize/')
+        createsvc = self.params.get("svcn", '/run/pool/createsvc/')
+
+        try:
+            # initialize a python pool object then create the underlying
+            # daos storage
+            self.pool = DaosPool(self.context)
+            self.pool.create(createmode, createuid, creategid,
+                             createsize, createsetid, None, None, createsvc)
+
+            pool_uuid = self.pool.get_uuid_str()
+            svc_list = ""
+            for i in range(createsvc):
+                svc_list += str(int(self.pool.svc.rl_ranks[i])) + ":"
+            svc_list = svc_list[:-1]
+
+            # running llnl
+            self.mpio.run_llnl(self.basepath, self.hostfile_clients, pool_uuid,
+                               llnl_test_repo)
+
+            # Parsing output to look for failures
+            # stderr directed to stdout
+            stdout = self.logdir + "/stdout"
+            searchfile = open(stdout, "r")
+            error_message = ["non-zero exit code", "MPI_Abort", "MPI_ABORT"]
+
+            for line in searchfile:
+                for i in range(len(error_message)):
```

Comment:
(pylint-consider-using-enumerate) Consider using enumerate instead of iterating with range and len